source/common/network/connection_impl.h

```diff
@@ -128,7 +129,7 @@ class ConnectionImpl : public ConnectionImplBase, public TransportSocketCallback
 
   TransportSocketPtr transport_socket_;
   ConnectionSocketPtr socket_;
-  StreamInfo::StreamInfoImpl stream_info_;
+  StreamInfo::StreamInfo& stream_info_;
```

Comment:
?

Reply:
TCP connection holds the unique pointer that it shares with this class. I'm fine re-structuring it if you prefer this connection to own the stream info.

Reply:
I see. What I would do here is make a `streamInfo()` virtual function that does the correct return based on the context. That is the typical pattern used.

Reply:
Thanks for explaining. Can you elaborate? I don't know which context the stream info off in ConnectionImpl. Looks like it only has a dispatcher and low-level connection and socket.

Reply:
Sorry this is OK. I forgot that there is no ServerConnectionImpl class, just ClientConnectionImpl.